### PR TITLE
Custom websites: sanitize input

### PR DESCRIPTION
**Pull request by lorenzoPrimi**
Fixes https://github.com/ooni/probe/issues/982

### CODE REVIEW

File: ooniprobe/View/Settings/CustomURLViewController.m [hunk 1]
```diff
@@ -44,8 +44,9 @@ -(IBAction)addRow:(id)sender{
 -(IBAction)run:(id)sender{
     urlArray = [[NSMutableArray alloc] init];
     for (NSString *url in self.urlsList){
-        if (![url isEqualToString:@"http://"] && ![url isEqualToString:@"https://"]){
-            Url *currentUrl = [Url checkExistingUrl:url];
+        NSString *sanitizedUrl= [url stringByTrimmingCharactersInSet:[NSCharacterSet newlineCharacterSet]];
+        if (![sanitizedUrl isEqualToString:@"http://"] && ![sanitizedUrl isEqualToString:@"https://"]){
```
Comment (bassosimone) [inline ooniprobe/View/Settings/CustomURLViewController.m]: This is slightly different from the Android algorithm. Maybe we should use the same algorithm in both cases? What is the reason why here you've introduced this extra check?

Comment (lorenzoPrimi) [reply ooniprobe/View/Settings/CustomURLViewController.m]: On iOS we have this nice `stringByTrimmingCharactersInSet` that takes a `NSCharacterSet` but on Android we are forced to use a regExp (that I don't personally like)

Comment (bassosimone) [reply ooniprobe/View/Settings/CustomURLViewController.m]: The question is why here you check for `http://` and `https://` and on Android you don't

Comment (lorenzoPrimi) [reply ooniprobe/View/Settings/CustomURLViewController.m]: On Android we use `Patterns.WEB_URL.matcher(value).matches()` that I believe does the same thing

Comment (bassosimone) [reply ooniprobe/View/Settings/CustomURLViewController.m]: you're right, thanks!

**Pull request merged**